sim/midas/src/main/scala/midas/passes/fame/FAMETransform.scala

```diff
@@ -178,6 +191,44 @@ object FAMEModuleTransformer {
         Connect(NoInfo, WSubField(WRef(buf), "I"), WRef(hostClock)),
         Connect(NoInfo, WSubField(WRef(buf), "CE"),
           And.reduce(Seq(WRef(enableReg), WRef(finishing), nReset)))))
+      // Add multicycle annotations
+      val clockName = info.name
+      val clockMFMR = clockChannel.clockMFMRMap(info)
+      val bufferOutputRT = mTarget.ref(buf.name).field("O")
+
+      if (currentModuleIsHub) {
+        // Leverage the MFMR hint provided by the clock bridge to relax the setup constraints on 
+        // intra-domain paths. Do this only for the hub, since it it is the only multiclock model.
+        //
+        // Using this multicycle setup constraint is conservative, since all
+        // inter-clock paths must still close timing at single a host cycle of
+        // delay. If we instead defined these generated clocks by specifying them
+        // as _divisions_ of the host_clock, the timer may give more margin to a
+        // path spanning two slower clock domains, when in reality they must meet
+        // a single host-cycle constraint.  For a counterexample, consider three
+        // clocks with periods of 2, 3, 4. This produces the following clock token schedule:
+        //
+        // token  : 0 1 2 3 4 5
+        // ====================
+        // clk 2  : 1 1 0 1 1 1
+        // clk 3  : 1 0 1 0 1 0
+        // clk 4  : 1 0 0 1 0 1
+        // t_time : 0 2 3 4 6 8
+        //
+        // While the latter two clocks have MFMRs of 2, they sometimes fire in
+        // back-to-back host-cycles (tokens 2->3).
+        //
+        // Note: "host_clock" is defined statically in the shell XDC.
+        val xdcAnno = XDCAnnotation(
+          XDCFiles.Implementation,
```

Comment:
Does this mean that only the Implementation XDC will have these constraints?  If so, I'm concerned that the retiming that happens during CL synthesis will continue to spend resources retiming paths in the composible cache that don't need to be retimed and then not have resources available for retiming.  Why wouldn't you use 
```suggestion
          XDCFiles.allFiles,
```

Reply:
So, this is a good question. 

I can put them in synthesis, and they should carry though. Or yeah, i could duplicate them. 

I picked just implementation because I had some trouble getting the synthesis constraints to apply correctly in synthesis, and most of the benefit is materialized in implementation anyways -- frankly, i was feeling lazy. 

I'll try again now.

Reply:
Ok.  I wasn't sure whether applying them in synthesis would mean they would come through the checkpoint and linking to be included in implementation.  I'll pick this up and also try it out.

Reply:
Yeah, you can choose the behavior you want.